test/IonWriter.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2012 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *  
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {assert} from "chai";
+import * as ion from "../src/IonTests";
+import {Decimal, IonType, IonTypes, Reader, Timestamp} from "../src/IonTests";
+import JSBI from "jsbi";
+
+// This dummy impl exists as a substitute for reflection directly against
+// the Writer interface (which TypeScript doesn't currently support)
+class TestWriter implements ion.Writer {
```

Comment:
This file/class might be worth naming something more specific. `UndefinedValueWriter`?

Reply:
Used generic terms with the idea that other tests against the `Writer` interface could go here, but I've renamed the file to `IonWriterUndefinedParameters`, and renamed `TestWriter` to `NoopWriter`.  Hope that helps a bit